tensorboard/data/experimental/base_experiment.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Base Class of Experiment Data Access API."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import abc
+
+import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BaseExperiment(object):
+    """Base class for experiment data access."""
+
+    # TODO(cais): Add list_scalar_runs().
+    # TODO(cais): Add list_scalar_tags().
+
+    @abc.abstractmethod
+    def get_scalars(self, runs_filter=None, tags_filter=None, pivot=True):
```

Comment:
Prefer using `None` as a default argument value rather than `True`, and
resolving inside the function itself. See, e.g.:
<https://unhashable.com/always-use-none-for-default-args-in-python/>